include/envoy/network/address.h

```diff
@@ -125,5 +125,104 @@ class Instance {
 
 typedef std::shared_ptr<const Instance> InstancePtr;
 
+/*
+ * Parse an internet host address (IPv4 or IPv6) and create an Instance from it.
+ * The address must not include a port number.
+ * @param ipAddr string to be parsed as an internet address.
+ * @return pointer to the Instance, or nullptr if unable to parse the address.
+ */
+InstancePtr parseInternetAddress(const std::string& ipAddr);
```

Comment:
There are somewhat related functions here: https://github.com/lyft/envoy/blob/master/source/common/network/utility.h#L50

If you want to move all address stuff into its own utility class/namespace that's cool with me, but could we group them together and put in source since not an abstract interface.